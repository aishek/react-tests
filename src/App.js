import React, { Component } from 'react';
import Tabs from "./Tabs"

class App extends Component {
  render() {
    return <Tabs/>
  }
}

export default App;
