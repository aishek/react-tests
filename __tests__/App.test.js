import React from 'react'
import App from '../src/App'
import { mount } from 'enzyme'

it('renders correctly', () => {
  const tree = mount(<App />)
  expect(tree).toMatchSnapshot()
})

it('switch tabs correctly', () => {
  const tree = mount(<App />)

  const tab = tree.find('[data-test-name="second-tab"]')
  tab.simulate('click')

  expect(tree).toMatchSnapshot()
})
