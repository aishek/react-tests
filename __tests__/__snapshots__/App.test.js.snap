// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<App>
  <_default>
    <Tabs
      defaultFocus={false}
      defaultIndex={null}
      forceRenderTabPanel={false}
      selectedIndex={null}
    >
      <UncontrolledTabs
        className="react-tabs"
        focus={false}
        forceRenderTabPanel={false}
        onSelect={[Function]}
        selectedIndex={0}
      >
        <div
          className="react-tabs"
          data-tabs={true}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <TabList
            className="react-tabs__tab-list"
            key=".0"
          >
            <ul
              className="react-tabs__tab-list"
              role="tablist"
            >
              <Tab
                className="react-tabs__tab"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-0"
                key=".0"
                panelId="react-tabs-1"
                selected={true}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-1"
                  aria-disabled="false"
                  aria-selected="true"
                  className="react-tabs__tab react-tabs__tab--selected"
                  id="react-tabs-0"
                  role="tab"
                  tabIndex="0"
                >
                  Title 1
                </li>
              </Tab>
              <Tab
                className="react-tabs__tab"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-2"
                key=".1"
                panelId="react-tabs-3"
                selected={false}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-3"
                  aria-disabled="false"
                  aria-selected="false"
                  className="react-tabs__tab"
                  id="react-tabs-2"
                  role="tab"
                  tabIndex={null}
                >
                  <span
                    data-test-name="second-tab"
                  >
                    Title 2
                  </span>
                </li>
              </Tab>
            </ul>
          </TabList>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-1"
            key=".1"
            selected={true}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-0"
          >
            <div
              aria-labelledby="react-tabs-0"
              className="react-tabs__tab-panel react-tabs__tab-panel--selected"
              id="react-tabs-1"
              role="tabpanel"
            >
              <h2>
                Any content 1
              </h2>
            </div>
          </TabPanel>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-3"
            key=".2"
            selected={false}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-2"
          >
            <div
              aria-labelledby="react-tabs-2"
              className="react-tabs__tab-panel"
              id="react-tabs-3"
              role="tabpanel"
            />
          </TabPanel>
        </div>
      </UncontrolledTabs>
    </Tabs>
  </_default>
</App>
`;

exports[`switch tabs correctly 1`] = `
<App>
  <_default>
    <Tabs
      defaultFocus={false}
      defaultIndex={null}
      forceRenderTabPanel={false}
      selectedIndex={null}
    >
      <UncontrolledTabs
        className="react-tabs"
        focus={false}
        forceRenderTabPanel={false}
        onSelect={[Function]}
        selectedIndex={1}
      >
        <div
          className="react-tabs"
          data-tabs={true}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <TabList
            className="react-tabs__tab-list"
            key=".0"
          >
            <ul
              className="react-tabs__tab-list"
              role="tablist"
            >
              <Tab
                className="react-tabs__tab"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-4"
                key=".0"
                panelId="react-tabs-5"
                selected={false}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-5"
                  aria-disabled="false"
                  aria-selected="false"
                  className="react-tabs__tab"
                  id="react-tabs-4"
                  role="tab"
                  tabIndex={null}
                >
                  Title 1
                </li>
              </Tab>
              <Tab
                className="react-tabs__tab"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-6"
                key=".1"
                panelId="react-tabs-7"
                selected={true}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-7"
                  aria-disabled="false"
                  aria-selected="true"
                  className="react-tabs__tab react-tabs__tab--selected"
                  id="react-tabs-6"
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    data-test-name="second-tab"
                  >
                    Title 2
                  </span>
                </li>
              </Tab>
            </ul>
          </TabList>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-5"
            key=".1"
            selected={false}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-4"
          >
            <div
              aria-labelledby="react-tabs-4"
              className="react-tabs__tab-panel"
              id="react-tabs-5"
              role="tabpanel"
            />
          </TabPanel>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-7"
            key=".2"
            selected={true}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-6"
          >
            <div
              aria-labelledby="react-tabs-6"
              className="react-tabs__tab-panel react-tabs__tab-panel--selected"
              id="react-tabs-7"
              role="tabpanel"
            >
              <h2>
                Any content 2
              </h2>
            </div>
          </TabPanel>
        </div>
      </UncontrolledTabs>
    </Tabs>
  </_default>
</App>
`;
